playbooks/provision.yml

```diff
@@ -0,0 +1,21 @@
+---
+- name: "Playbook to provision Argo into a single namespace."
+
+  hosts: localhost
+  connection: local
+
+  vars_files:
```

Comment:
I'm unsure if you need to include them explicitly @harshad16 ?

Reply:
If the variable is in *group_vars*, then it does not need to be explicitly added in the playbook. The only requirement is to have the playbook and group_vars on the same directory level.

Reply:
You have to include them specificly if you don't want to use all of them.

Reply:
yes, sure. I don't see any other group_vars though.
Theoretically, group_vars are dependent on the host's groups as we don't use the host's groups, we put out variables in group_vars/all, however as you said if we need specific ones we should explicitly state them, that true. The recommended way is to have the vars in the role if we need them specifically for them.
Just my thoughts, the use of vars_files is good method indeed as well.

Reply:
Well ye, we cannot use the `group_vars/all` because we don't wanna use them while deprovisioning. We could, of course, move the vars directly to the playbook if we don't want to have them in a separate file but since we have the **vault**, we wouldn't get rid of that denotation anyway.